### PR TITLE
[SDXL Refiner] Fix refiner forward pass for batched input

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

As found by @bghira in #4297 there was a small bug for the refiner when the batch size > 1.
For certain use cases (low steps, high precision) the bug led to serious artifacts as seen in #4297 

We also adapt the tests here to make sure the refiner is correctly tested going forward.

### CODE REVIEW

File: tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl_inpaint.py [hunk 1]
```diff
@@ -33,7 +33,6 @@
     UniPCMultistepScheduler,
 )
 from diffusers.utils import floats_tensor, torch_device
-from diffusers.utils.testing_utils import enable_full_determinism, require_torch_gpu
```
Comment (patrickvonplaten) [inline tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl_inpaint.py:36]:
```suggestion
from diffusers.utils.testing_utils import enable_full_determinism, require_torch_gpu
```

File: src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_img2img.py [hunk 2]
```diff
@@ -906,15 +906,17 @@ def denoising_value_valid(dnv):
             negative_aesthetic_score,
             dtype=prompt_embeds.dtype,
         )
+        add_time_ids = add_time_ids.repeat(batch_size * num_images_per_prompt, 1)
```
Comment (bghira) [inline src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_img2img.py:909]: those blasted timestep embeds!

**Comment by bghira** [comment]
there's now minimal differences in the outputs for bs1 and bs2 :) 

image
image

bonus, we can now see the kid's face a bit better:
image

**Comment by bghira** [comment]
Ran a batch size of 2 with the invisible-watermark, eg. this is do_latents = False, just to ensure it's not a problem.

image
image

**Review by bghira — APPROVED** [review]

**Comment by patrickvonplaten** [comment]
Hmm sadly the email address doesn't seem to work for co-authorship. Hope the PR description is enough credit here :-)

**Comment by patrickvonplaten** [comment]
Merging! Many thanks for bearing with me yesterday evening @bghira !

**Comment by bghira** [comment]
the fact it means i wasnt crazy, meant the world to me ahaha

**Pull request merged**